src/transaction_processor.py

```
@@ -0,0 +1,182 @@
+            )
+            logger.debug("Successfully deleted entries for block number: %s", max_block)
+
+            return max_block
+        except Exception as e:
+            logger.error("Error accessing database: %s", e)
+            return self.blockchain_data.get_latest_block()
+
+    def process(self, start_block: int):
+        """Main Daemon loop that finds imbalances for txs and prices."""
+        previous_block = start_block
+        unprocessed_txs: List[Tuple[str, int, int]] = []
+        logger.info("%s Daemon started. Start block: %d", self.chain_name, start_block)
+
+        while True:
+            try:
+                latest_block = self.blockchain_data.get_latest_block()
+                print(previous_block, latest_block)
+                new_txs = self.blockchain_data.fetch_tx_data(
+                    previous_block, latest_block
+                )
+                all_txs = new_txs + unprocessed_txs
+                unprocessed_txs.clear()
+
+                for tx_hash, auction_id, block_number in all_txs:
+                    try:
+                        self.process_single_transaction(
+                            tx_hash, auction_id, block_number
+                        )
+                    except Exception as e:
+                        unprocessed_txs.append((tx_hash, auction_id, block_number))
+                        logger.error(f"Error processing transaction {tx_hash}: {e}")
+
+                previous_block = latest_block + 1
+                time.sleep(CHAIN_SLEEP_TIME)
+
+            except Exception as e:
+                logger.error(f"Error in processing loop: {e}")
+                time.sleep(CHAIN_SLEEP_TIME)
+
+    def process_single_transaction(
+        self, tx_hash: str, auction_id: int, block_number: int
+    ):
+        """Function processes a single tx to find imbalances, prices."""
+        token_imbalances = self.imbalances.compute_imbalances(tx_hash)
```

Instead of making `compute_imbalances` return None, I would raise an error there. That error can then be caught here if desired.

At the moment, the code just raises an error if `token_imbalances is None` so it is equivalent to just raising an error in `compute_imbalances` and not caching it here.